code/actions/UnpublishItemWorkflowAction.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * Unpublishes an item
+ *
+ * @author     marcus@silverstripe.com.au
+ * @license    BSD License (http://silverstripe.org/bsd-license/)
+ * @package    advancedworkflow
+ * @subpackage actions
+ */
+class UnpublishItemWorkflowAction extends WorkflowAction {
+
+	private static $db = array(
+		'PublishDelay' => 'Int'
+	);
+
+	public static $icon = 'advancedworkflow/images/publish.png';
+
+	public function execute(WorkflowInstance $workflow) {
+		if (!$target = $workflow->getTarget()) {
+			return true;
+		}
+
+		if (class_exists('AbstractQueuedJob') && $this->PublishDelay) {
+			$job   = new WorkflowPublishTargetJob($target);
```

Comment:
This will actually cause the item to be published - should it actually pass in "unpublish" as the second parameter to the job? Also, can it be 'UnpublishDelay' instead?